Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs

```diff
@@ -30,65 +39,73 @@ public async UniTask<Texture2D> TakeAndSaveScreenshotAsync(CharacterPreviewContr
+                    mipCount = 1, useMipMap = false
+                };
 
-                // Create the final Texture2D with the correct small dimensions.
-                var screenshotTexture = new Texture2D(targetWidth, targetHeight, TextureFormat.RGBA32, false);
+                downscaledSRGB = RenderTexture.GetTemporary(desc);
 
-                // Set the temporary RT as active and read its (now small) contents.
-                RenderTexture.active = tempRT;
-                screenshotTexture.ReadPixels(new Rect(0, 0, targetWidth, targetHeight), 0, 0);
+                // GPU downscale + sRGB encode in one pass
+                Graphics.Blit(source, downscaledSRGB);
 
-                var pixels = screenshotTexture.GetPixels();
-                for (int p = 0; p < pixels.Length; p++)
+                // Async readback: avoids render-thread stalls, completion occurs on a later frame
+                var req = await AsyncGPUReadback.Request(downscaledSRGB).WithCancellation(ct);
+                if (req.hasError)
                 {
-                    pixels[p] = pixels[p].gamma;
+                    ReportHub.LogError(ReportCategory.OUTFITS, "GPU readback failed.");
+                    return null;
                 }
 
-                screenshotTexture.SetPixels(pixels);
-                
-                
-                screenshotTexture.Apply();
+                ct.ThrowIfCancellationRequested();
 
-                // Clean up: release the temporary RT and reset the active one.
-                RenderTexture.active = null;
-                RenderTexture.ReleaseTemporary(tempRT);
+                var srgbNative = req.GetData<byte>();
 
-                ct.ThrowIfCancellationRequested();
+                byte[] pngBytes;
+                // Encode PNG directly from the readback buffer
+                // (no Texture2D.Get/SetPixels, no CPU gamma conversion)
+                using (var pngNative = ImageConversion.EncodeNativeArrayToPNG(
+                           srgbNative, downscaledSRGB.graphicsFormat, (uint)w, (uint)h))
+                {
+                    pngBytes = pngNative.ToArray();
+                }
 
-                byte[] pngBytes = screenshotTexture.EncodeToPNG();
+                // Persist to disk on a background thread
                 string filePath = GetFilePathForSlot(userId, slotIndex);
+                Directory.CreateDirectory(Path.GetDirectoryName(filePath)!);
 
-                if (filePath != null)
-                    Directory.CreateDirectory(Path.GetDirectoryName(filePath));
-                else return null;
-
+                await UniTask.SwitchToThreadPool();
                 await File.WriteAllBytesAsync(filePath, pngBytes, ct);
+                await UniTask.SwitchToMainThread(ct);
 
-                return screenshotTexture;
+                // Return bytes so UI can display thumbnail
+                // immediately (no disk read required)
+                return pngBytes;
```

Comment:
I suppose you can use the native array directly, moreover if it's just to create texture you can use `Graphics.Copy` from `downscaledSRGB` directly to the 2d texture without a need to decode again